packages/chat-component-bindings/src/messageThreadSelector.ts

```diff
@@ -119,30 +146,50 @@ const hasValidParticipant = (chatMessage: ChatMessageWithStatus): boolean =>
  * @public
  */
 export const messageThreadSelector: MessageThreadSelector = createSelector(
-  [getUserId, getChatMessages, getLatestReadTime, getIsLargeGroup],
-  (userId, chatMessages, latestReadTime, isLargeGroup) => {
+  [getUserId, getChatMessages, getLatestReadTime, getIsLargeGroup, getReadInformation, getParticipants],
+  (userId, chatMessages, latestReadTime, isLargeGroup, readInformation, participants) => {
+    // get number of participants
+    const messageThreadParticipantCount = Object.values(participants).length - 1;
+    // for each chat message, add read receipt information when readInformation.id === chatmessage.id
+    const convertedChatMessages = Object.values(chatMessages).map(function (message) {
+      const readReceipt = Object.values(readInformation).filter((info) => info.chatMessageId === message.id);
+      // readReceipt is getting duplicate information, remove duplicate here
+      const uniqueReadReceipt = readReceipt.filter(
+        (value, index, self) =>
+          index ===
+          self.findIndex(
+            (t) =>
+              toFlatCommunicationIdentifier(t.sender) === toFlatCommunicationIdentifier(value.sender) &&
+              t.chatMessageId === value.chatMessageId
+          )
+      );
+      const convertedMessage = { message, readReceipt: uniqueReadReceipt };
```

Comment:
And readReceipts + uniqueReadReceipts as awalys

Reply:
Updated PR